ipamd/rpc_handler.go

```diff
@@ -97,6 +97,14 @@ func (s *server) DelNetwork(ctx context.Context, in *pb.DelNetworkRequest) (*pb.
 		ip, deviceNumber, err = s.ipamContext.dataStore.UnassignPodIPv4Address(&k8sapi.K8SPodInfo{
 			Name:      in.K8S_POD_NAME,
 			Namespace: in.K8S_POD_NAMESPACE})
+
+		// if a pod gets scheduled to a node when the aws cni is restarting,
+		// kubelet would kill and recreate pod. When kubelet tries to kill
+		// the container initially, this block would make sure the ipamD agent would
+		// make sure we dont return an error.
+		if err == datastore.ErrUnknownPod {
+			err = nil
+		}
```

Comment:
I guess you were basing this change on tests done on v1.5.3 or earlier? This kind of change was the main cause of issues in v1.5.4. It looks innocent, but had other effects because of the cleanup that got triggered in the CNI binary. See #641 for some details.

Also, #688 should fix this issue to stop returning an error to Kubelet when we try to delete a pod that failed to get an IP. That change should be in the v1.6.0-rc4 release.

Reply:
Thanks @mogren. That makes sense.